### PR TITLE
feat(observer): add subject snprintf

### PR DESCRIPTION
**Pull request by Pavi92**
snprintf for lv_subject (LV_SUBJECT_TYPE_STRING)

### CODE REVIEW

File: src/others/observer/lv_observer.h [hunk 1]
```diff
@@ -117,6 +117,13 @@ void lv_subject_init_string(lv_subject_t * subject, char * buf, char * prev_buf,
  */
 void lv_subject_copy_string(lv_subject_t * subject, const char * buf);
 
+/**
+ * Print a formated string to a subject. It will notify all the observers as well.
+ * @param subject   pointer to the subject
+ * @param buf       the format string
+ */
+void lv_subject_snprintf(lv_subject_t * subject, const char * format, ...);
```
Comment (liamHowatt) [inline src/others/observer/lv_observer.h]:
```suggestion
 * @param format    the format string
 */
void lv_subject_snprintf(lv_subject_t * subject, const char * format, ...) LV_FORMAT_ATTRIBUTE(2, 3);
```

**Review by liamHowatt — DISMISSED** [review]
Thanks for the feature!

**Review by kisvegabor — COMMENTED** [review]
Thank you! 

Please also add a simple test [here](https://github.com/lvgl/lvgl/blob/master/tests/src/test_cases/test_observer.c).

**Comment by Pavi92** [comment]
@kisvegabor done

**Review by kisvegabor — APPROVED** [review]
Nice, thank you!

**Review by liamHowatt — APPROVED** [review]

**Pull request merged**